File: models_sql/mlModelWeights.js
module.exports = (sequelize, DataTypes) => {

const MlModelWeights = sequelize.define('MlModelWeights', {
    id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
    },
    featureImportanceScores: {
        type: DataTypes.JSON,
        allowNull: true
    },
    hiddenToOutputWeights: {
        type: DataTypes.JSON,
        allowNull: true
    },
    inputToHiddenWeights: {
        type: DataTypes.JSON,
        allowNull: true
    },
    sport:{
        type: DataTypes.INTEGER,
        allowNull: false,
        references: {
            model: 'Sports',
            key: 'id'
        }
    }
}, 
{
    tableName: 'MlModelWeights',
    timestamps: true,
    indexes: [
        {
            unique: true,
            fields: ['sport']
        }
    ]
});

MlModelWeights.associate = (models) => {
    MlModelWeights.belongsTo(models.Sports, {
        foreignKey: 'sport',
        as: 'MlModelWeights'
    })
}

return MlModelWeights;
}

File: app/database/mlModelWeights.py
from sqlalchemy import String, Integer, DateTime, Boolean, Float, JSON, Index, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column, relationship
from .base import Base


class MlModelWeights(Base):
    __tablename__ = "MlModelWeights"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    featureImportanceScoresTeam: Mapped[JSON] = mapped_column(JSON, nullable=True)
    featureImportanceScoresFull: Mapped[JSON] = mapped_column(JSON, nullable=True)
    hiddenToOutputWeights: Mapped[JSON] = mapped_column(JSON, nullable=False)
    inputToHiddenWeights: Mapped[JSON] = mapped_column(JSON, nullable=False)
    sport: Mapped[int] = mapped_column(ForeignKey("Sports.id"), unique=True)

    #relationships
    # sport: Mapped["Sports"] = relationship("Sports", foreign_keys=[sport], back_populates="mlModelWeights")


    # __table_args__ = (
    #     Index('ix_sport', 'sport', unique=True),
    # )   